vocode/streaming/agent/state_agent.py

```diff
@@ -1593,8 +1637,18 @@ async def call_ai(self, prompt, tool=None, stop=None, stream_output=True):
                 self.logger.info(
                     f"Streaming final message. Raw: {buffer}  interpolated: {interpolated_content}"
                 )
+                # Need to pass the same object to the json transcript and the produce_interruptible_agent_response_event_nonblocking
+                transcript_message = StateAgentTranscriptMessage(
+                    role="message.bot",
+                    message=interpolated_content,
+                    message_sent="",
+                )
+                messages_to_add.append(transcript_message)
```

Comment:
save the mutable object to a list so we can update the message sent in send speech to output. `messages_to_add` are eventually passed to update_history.